Chatkit/Auth/TokenProvider.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  TokenProvider.swift
+//  Chatkit
+//
+//  Created by Mike Pye on 06/12/2019.
+//  Copyright © 2019 Pusher Ltd. All rights reserved.
+//
+// STUB FILE. This file demonstrates the desired interface, but contains to implementation
+// TODO: Move many of these types to PusherPlatform
+
+import Foundation
+import PusherPlatform
+
+// TODO: Temporary
+extension String: Error { }
+
+// Re-export so that PusherPlatform import is not required in client code
+public typealias TokenProvider = PPTokenProvider
+public typealias TokenProviderResult = PPTokenProviderResult
+
+/// A factory for `TokenProvider`s.
+///
+/// The methods on this object create token provider instances which can interface with an HTTP endpoint in your backend, or the Chatkit Test Token Provider (for use during development)
+public struct ChatkitTokenProviders {
+    /// Creates a `TokenProvider` which fetches tokens from the Chatkit Test Token Provider. It should be used for development purposes only.
+    ///
+    /// The Chatkit Test Token Provider is a token provider endpoint hosted by Pusher which returns tokens for the requested `userId` without authenticating the user. It is intended for development convenience and *must not be enabled in instances which host production data*.
+    ///
+    /// The test token provider must be enabled for your instance in the Chatkit dashboard.
+    ///
+    /// - Parameters:
+    ///     - instanceLocator: The locator string for your instance, which you also provide when initialising the `Chatkit` object.
+    ///     - userId: The user identifier to fetch a token for. The provider will sign a valid token for any user identifier requested, without applying any kind of authentication.
+    /// - Returns:a `TokenProvider` which fetches tokens from the Chatkit Test Token Provider.
+    public static func chatkitTestTokenProvider(instanceLocator: String, userId: String) -> TokenProvider {
```

Comment:
Wouldn't it be better to have `create` prefix so that it's clear it's a factory method and each call will create a new instance?

Reply:
That would be consistent with factory methods we already have in `Chatkit` class (entry point).

Reply:
https://swift.org/documentation/api-design-guidelines/#strive-for-fluent-usage

The official Swift style guide says they should all be called `make...`

Reply:
Should we rename `Chatkit` methods in that case?

Reply:
I'll make a separate PR to change them all...

Reply:
Doesn't feel like matching mental model of an Android dev so probably best to have it as `create` there.

Reply:
Yes, we will follow the appropriate conventions for each ecosystem.